### DISCUSSION TITLE
GACOS projection in LOS direction

### DISCUSSION THREAD

**Discussion by sincaliz** [gmtsar/gmtsar#991] [Q&A]
Hello all,

I would like to apply GACOS corrections to my interferograms to remove tropospheric noise. GACOS products give the Zenith Total Delay maps which imply I have to project them into the LOS direction to remove them from my phase data. I have some questions:

**1.** In the ALOS metadata I have the off_nadir_angle. I found that 90-off_nadir_angle=incidence_angle. Could this unique value be used for the GACOS projection in LOS direction?, or

**2.** I can estimate the LOS vector (de, dn, du) per pixel using the SAT_look function. I use the longitude, latitude coordinates from my unwrap_mask_ll.grd file and the elevations from the topo_ra_ll.grd to create the input file. I did this for each interferogram using the respective master.prm file. From the vector (de, dn, du) I can estimate the incidence angle, but it is variable across the interferogram. To project the GACOS ZTD, should I resample it to the interferogram dimensions and resolution, and then use each pixel incidence angle? Another question on this point. Is it recommended to use the unwrap_mask_ll.grd and topo_ra_ll.grd or better to use the non-geocoded phase and dem.grd or topo_ra.grd? 

**3.** From the results in **2** I can obtain an average incidence angle. Is it OK to use this unique value for the GACOS projection ? What if it is slightly different from the one in **1**

I am looking forward your help regarding the best approach to follow.

Thanks in advance

**Comment by Xiaohua-Eric-Xu** [comment]
> Hello all,
> 
> I would like to apply GACOS corrections to my interferograms to remove tropospheric noise. GACOS products give the Zenith Total Delay maps which imply I have to project them into the LOS direction to remove them from my phase data. I have some questions:
> 
> **1.** In the ALOS metadata I have the off_nadir_angle. I found that 90-off_nadir_angle=incidence_angle. Could this unique value be used for the GACOS projection in LOS direction?, or

This is doable when you have a narrow swath width, but if the look angle changes a lot across your entire swath, 2 is better. 

> **2.** I can estimate the LOS vector (de, dn, du) per pixel using the SAT_look function. I use the longitude, latitude coordinates from my unwrap_mask_ll.grd file and the elevations from the topo_ra_ll.grd to create the input file. I did this for each interferogram using the respective master.prm file. From the vector (de, dn, du) I can estimate the incidence angle, but it is variable across the interferogram. To project the GACOS ZTD, should I resample it to the interferogram dimensions and resolution, and then use each pixel incidence angle? Another question on this point. Is it recommended to use the unwrap_mask_ll.grd and topo_ra_ll.grd or better to use the non-geocoded phase and dem.grd or topo_ra.grd?

You could convert du to grids with `xyz2grd` and then `grdsample`, or just directly use `surface`. After that you should have a full resolution look angle vector grid. Usually the look angle may change a bit locally when there are really steep topography. If not, they are smooth enough to be interpolated. 

> **3.** From the results in **2** I can obtain an average incidence angle. Is it OK to use this unique value for the GACOS projection ? What if it is slightly different from the one in **1**
>
> I am looking forward your help regarding the best approach to follow.
> 
> Thanks in advance
